eclipse-che/che-operator
# chore: Propagate Bitbucket SAAS Oauth secret properties

Signed-off-by: Igor Vinokur <ivinokur@redhat.com>



### What does this PR do?
Add Bitbucket Oauth2 properties support. Propagate Bitbucket SAAS client id and client secret Oauth2 properties as workpsace environment variables.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21517

### How to test this PR?


```bash
cat EOF << > /tmp/patch.yaml
<PATCH_CONTENT>
EOF

chectl server:deploy \
     --installer operator \
     --platform <PLATFORM_TO_DEPLOY> \
     --che-operator-image <CUSTOM_OPERATOR_IMAGE> \
     --che-operator-cr-patch-yaml /tmp/patch.yaml
```
Create bitbucket oauth secret in the che-server namespace:
```yaml
kind: Secret
apiVersion: v1
metadata:
  name: bitbucket-oauth-config
  namespace: eclipse-che
  labels:
    app.kubernetes.io/component: oauth-scm-configuration
    app.kubernetes.io/part-of: che.eclipse.org
  annotations:
    che.eclipse.org/oauth-scm-server: bitbucket
data:
  id:<base 64 encoded client id from the bitbucket oauth consummer>
  secret: <base 64 encoded secret from the bitbucket oauth consummer>
type: Opaque
```
Start a workspace and see the Bitbucket Oauth2 environment variable 
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
